### PR TITLE
 Delete terms on plugin uninstall

### LINKED ISSUE #8452: Delete terms on plugin uninstall
## Feature Description

When the plugin is uninstalled, all terms from the `web_story_media_source` taxonomy should be deleted.

[Example](https://github.com/ampproject/amp-wp/blob/a99fb52013d2de8b8b23d10955e7d8e750101222/includes/uninstall-functions.php#L91-L122) from amp plugin. 

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
When the plugin is uninstalled, all terms from the web_story_media_source taxonomy should be deleted.

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8452

### CODE REVIEW

File: includes/uninstall.php [hunk 1]
```diff
@@ -161,6 +161,34 @@ function delete_posts() {
...
 }
+
+/**
+ * Deletes all media source terms.
+ *
+ * @since 1.10.0
+ *
+ * @return void
+ */
+function delete_terms() {
```
Comment (spacedmonkey) [inline includes/uninstall.php:173]: This is different from the AMP plugin's version for two reason. 
1. There are much less terms in this taxonomy, as they are created in code. 
2. PHPStan, PHPMD did not like the amp version.

File: includes/uninstall.php [hunk 2]
```diff
@@ -161,6 +161,34 @@ function delete_posts() {
...
+			'hide_empty' => false,
+			'fields'     => 'ids',
+		]
+	);
+	if ( empty( $term_ids ) || ! is_array( $term_ids ) ) {
+		return;
+	}
+
+	foreach ( $term_ids as $term_id ) {
+		// @phpstan-ignore-next-line
```
Comment (spacedmonkey) [inline includes/uninstall.php]: PHPStan is wrong here.

Comment (swissspidy) [reply includes/uninstall.php]: What error exactly is being ignored here?

File: includes/uninstall.php (hunk 1 above)
Comment (swissspidy) [reply includes/uninstall.php:173]: What's the difference then to the version from the AMP plugin?

And what did PHPStan & PHPMD not like about it exactly?

File: includes/uninstall.php [hunk 3]
```diff
@@ -161,6 +161,34 @@ function delete_posts() {
...
+	$taxonomy = Media::STORY_MEDIA_TAXONOMY;
+
+	$term_ids = get_terms(
+		[
+			'taxonomy'   => Media::STORY_MEDIA_TAXONOMY,
+			'hide_empty' => false,
+			'fields'     => 'ids',
+		]
+	);
+	if ( empty( $term_ids ) || ! is_array( $term_ids ) ) {
```
Comment (swissspidy) [inline includes/uninstall.php:183]: This should have the same effect:

```suggestion
	if ( ! is_array( $term_ids ) ) {
```

Comment (spacedmonkey) [reply includes/uninstall.php:183]: What if the array is empty?

Comment (swissspidy) [reply includes/uninstall.php:183]: Then the foreach loop below will not do anything

Comment (spacedmonkey) [reply includes/uninstall.php:183]: If that is the case, then why not return early then?

Comment (swissspidy) [reply includes/uninstall.php:183]: It feels cleaner and is less code. But I'm indifferent.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**